ppl/src/main/antlr/OpenSearchPPLParser.g4

```diff
@@ -315,7 +323,8 @@ tableSourceClause
 
 // join
 joinCommand
-   : (joinType) JOIN sideAlias joinHintList? joinCriteria? right = tableOrSubqueryClause
+   : joinType JOIN sideAlias joinHintList? joinCriteria right = tableOrSubqueryClause
```

Comment:
We should expose only one syntax at a time? Concern of enable both PPL and SPL-compatible syntax is 
* Hard to document and maintain, tooling (like syntax highlighting, query editors) must also be mode-aware.
* Increases complexity of semantic validation and testing.

Reply:
The current design is when the config is set to true, both ppl and spl syntax are supported. For your concerns, let me check the challenge in implementation level next week. I don’t think the document has any blocker. In future we can add a legacy config to disable the current ppl syntax as long as the implementation level has no blocker. For join, the current ppl join is a mutation of join of spl2. We may still need to support since splunk can both support spl1 and spl2. For your second concern, as long as implementation part has no blocker, the validation should work well since validation is done by antlr4 itself.